### DISCUSSION TITLE
Auto Suggest Box, show suggestions on mouse focus

### DISCUSSION THREAD

**Discussion by ShehabFekry** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3648] [Q&A]
I'm using a data grid that has an Auto Suggest Box column everything is working fine so far, i want to make the box show the list all the suggestions when the user focuses the box and it has an empty string to help him select 
i tried this event but it doesn't do anything while it's empty, 

i wanted to do the same when the user loses focus of combo box if he edited that value like for example the suggestion is "Active Storage" then the user selects its and edit it to "Active Storageeee" it stores this value while its not on the options, i handled this using LostFocus event but i was wondering if there's a property to handle this as well

```
 <materialDesign:AutoSuggestBox DropDownElevation="Dp0" Text="{Binding SelectedType, 
 UpdateSourceTrigger=PropertyChanged, Mode=TwoWay}" Suggestions="{Binding SpaceTypes}" ValueMember="Key">
                                                <i:Interaction.Triggers>
                                                    <i:EventTrigger EventName="GotFocus">
                                                        <i:InvokeCommandAction Command="{Binding FocusCommand}" />
                                                    </i:EventTrigger>
                                                </i:Interaction.Triggers>
        private void Focus(object obj)
        {
                if (string.IsNullOrEmpty(SelectedType))
                {
                    SpaceTypes = new ObservableCollection<KeyValuePair<string, HVACLoadSpaceType>>(_originalTypes);
                }
        } 
```
   
   is there any property or event to handle this issue ?

**Comment by nicolaihenriksen** [comment]
@ShehabFekry Regarding the first issue, displaying the list of options when focused, is not something I believe you can configure it to today, but it seems like a relatively easy thing to add as an opt-in feature though. It seems like you are very strict on only wanting this behavior if there isn't any content already. I could see use-cases for both what you describe, but also always opening on focus regardless of content. So perhaps a `SuggestMode` property taking an `enum` value of `OpenOnEnteredText`, `OpenOnFocus`, `OpenOnFocusWithEmptyContent` (don't mind the naming, I am horrible at that 😆) or something along those lines could be useful.

Regarding your second issue where the user types in something not present in the list, I am not really sure I understand what the issue seems to be. Could you elaborate?
